### PR TITLE
[FIX] account{,edi_ubl_cii}: harmonize invoice upload

### PR DESCRIPTION
**Pull request by rigr-odoo**
# Aim of the PR
This PR aims at harmonizing the invoice upload in two cases:
- inside the Accounting app (Customer Invoice vs Vendor Bills)
- between the Accounting and the Documents apps

### Before this PR:
- Go to Accounting / Customer Invoices
- Upload a PDF with Factur-X XML embedded
- The document is uploaded as a Vendor Bill

### After this PR
This should not be the case anymore since we want to allow the import of customer invoices (that can be generated elsewhere)
This commit makes sure that we check from where we import the PDF (Upload button of Customer Invoices or Vendor Bills) to determine in which journal the generated invoice should go via the context.

## Between the Accounting and the Documents apps
To go even further, we also harmonize the invoice upload with the Documents app as such:

### Before this PR
We had two different behavior for the same functionality depending on the App used.
In the Documents App:
- Upload a PDF document with embedded Factur-X in the Documents app
- Click on Create Bill in the actions of the document
- The OCR is triggered.

In the Accounting App:
- Upload a PDF document with embedded Factur-X via the "Upload Bills" button in the Accounting Dashboard
- The invoice is created using the information contained in the Factur-X XML embedded in the PDF without using the OCR.

### After this PR
We should have the same behavior in both apps. Furthermore, if the invoice information can be retrieved from the XML, it is useless and more error-prone to use the OCR.
This commit fixes this by calling the Accounting importing function instead which also simplifies greatly the code

task-id 2961932

Enterprise PR: https://github.com/odoo/enterprise/pull/30879

### CODE REVIEW

File: addons/account/models/account_journal.py [hunk 1]
```diff
@@ -643,7 +643,13 @@ def create_invoice_from_attachment(self, attachment_ids=[]):
                 invoice = self.env['account.move'].create({})
             invoice.with_context(no_new_invoice=True).message_post(attachment_ids=[attachment.id])
             invoices += invoice
+        return invoices
 
+    def create_invoice_from_attachment(self, attachment_ids=None):
+        '''
+         :return: A action redirecting to account.move tree/form view.
```
Comment (william-andre) [inline addons/account/models/account_journal.py:650]: The docstring is more important for this compared to `_generate_invoice_from_attachment` as it is a public method

File: addons/account/models/account_journal.py [hunk 2]
```diff
@@ -622,10 +622,10 @@ def action_configure_bank_journal(self):
         # We simply call the setup bar function.
         return self.env['res.company'].setting_init_bank_account_action()
 
-    def create_invoice_from_attachment(self, attachment_ids=[]):
-        ''' Create the invoices from files.
-         :return: A action redirecting to account.move tree/form view.
-        '''
+    def _generate_invoice_from_attachment(self, attachment_ids=None):
```
Comment (william-andre) [inline addons/account/models/account_journal.py:625]: `_create_invoice_from_attachment` feels like a better name

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Comment by rigr-odoo** [comment]
@robodoo rebase-ff

**Comment by rdeodoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**